WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -90,4 +124,35 @@ class PostListMainViewModel @Inject constructor(
             }
         }
     }
+
+    sealed class AuthorFilterListItemUIState(
+        internal val text: UiString, @DrawableRes val iconRes: Int, val isSelected: Boolean
+    ) {
+        class Everyone(isSelected: Boolean) : AuthorFilterListItemUIState(
+                text = UiStringRes(R.string.post_list_author_everyone),
+                iconRes = R.drawable.ic_multiple_users_white_24dp,
+                isSelected = isSelected
+        )
+
+        class Me(val avatarUrl: String?, isSelected: Boolean) : AuthorFilterListItemUIState(
+                text = UiStringRes(R.string.post_list_author_me),
+                iconRes = R.drawable.ic_user_circle_grey_24dp,
+                isSelected = isSelected
+        )
+    }
+
+    enum class AuthorFilterSelection {
+        ME, EVERYONE;
+
+        companion object {
+            @JvmStatic
+            fun valueOf(value: String, default: AuthorFilterSelection): AuthorFilterSelection {
```

Comment:
Can we add a `Int` value to the `enum` and use that instead of the `String` representation of it. That way it'll be less likely to break if we refactor the names.

Reply:
Sure, I went with `.name` over `.ordinal` to avoid the issue of adding or rearranging items. However we could certainly change to an `id` instead.

Reply:
Resolved in 0da6871b3725a5f260ff79460d45623d61fa7bf3